src/pages/layout/Search/Index.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {useState, useEffect, useMemo} from "react";
+import {throttle} from "lodash";
+import {Autocomplete, AutocompleteInputChangeReason} from "@mui/material";
+import SearchTransaction from "./SearchTransaction";
+import SearchAccount from "./SearchAccount";
+import SearchInput from "./SearchInput";
+import ResultPaper from "./ResultPaper";
+import SearchResultNotFound from "./SearchResultNotFound";
+
+const HEX_REGEXP = /^(0x)?[0-9a-fA-F]+$/;
+
+type OptionType = React.ReactNode[];
+
+function getSearchResults(searchText: string): OptionType {
+  searchText = searchText.trim();
+
+  if (!HEX_REGEXP.test(searchText)) {
+    // if it's not hexadecimal characters, no results
+    return [<SearchResultNotFound />];
+  }
+
+  const results = [
+    <SearchTransaction txnHashOrVersion={searchText} />,
+    <SearchAccount address={searchText} />,
+  ];
+
+  if (results.length === 0) {
+    return [<SearchResultNotFound />];
+  } else {
+    return results;
+  }
+}
+
+export default function HeaderSearch() {
+  const [inputValue, setInputValue] = useState<string>("");
+  const [options, setOptions] = useState<OptionType>([]);
+
+  const fetch = useMemo(
+    () =>
+      throttle(
+        (searchText: string, callback: (results: OptionType) => void) => {
+          callback(getSearchResults(searchText));
+        },
+        200,
+      ),
+    [],
+  );
+
+  useEffect(() => {
+    let active = true;
+
+    if (inputValue === "") {
+      setOptions([]);
+      return;
+    }
+
+    fetch(inputValue, (results: OptionType) => {
+      if (active) {
+        setOptions(results);
+      }
+    });
+
+    return () => {
+      active = false;
+    };
+  }, [inputValue, fetch]);
+
+  // hide autocomplete dropdown until text has been entered
+  const [open, setOpen] = useState(false);
+
+  const handleInputChange = (
+    event: any,
+    newInputValue: React.SetStateAction<string>,
+    reason: AutocompleteInputChangeReason,
+  ) => {
+    if (newInputValue.length === 0) {
+      if (open) setOpen(false);
+    } else {
+      if (!open) setOpen(true);
+    }
+    if (event && event.type === "blur") {
+      setInputValue("");
+    } else if (reason !== "reset") {
```

Comment:
do we need to handle the `reset` reason case also...?

Reply:
TBH I copied this part directly from Max's old code and didn't look very deep into it. I'd trust what has been working well before I figure out otherwise.